### ISSUE TITLE
OS X 10.11: RunTimeError('no suitable implementation for this system')

### ISSUE THREAD

**Issue by lweijl** [atdt/monotonic#10]
Hi there,

Not sure if I should ask this here, but running the following command results in a 'no suitable implementation for this system' RunTimeError.

`
$ nova -h
Traceback (most recent call last):
  File "/usr/local/bin/nova", line 9, in <module>
    load_entry_point('python-novaclient==2.31.0', 'console_scripts', 'nova')()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 357, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 2394, in load_entry_point
    return ep.load()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 2108, in load
    entry = __import__(self.module_name, globals(),globals(), ['__name__'])
  File "/Library/Python/2.7/site-packages/python_novaclient-2.31.0-py2.7.egg/novaclient/__init__.py", line 17, in <module>
    from novaclient import api_versions
  File "/Library/Python/2.7/site-packages/python_novaclient-2.31.0-py2.7.egg/novaclient/api_versions.py", line 26, in <module>
    from novaclient import utils
  File "/Library/Python/2.7/site-packages/python_novaclient-2.31.0-py2.7.egg/novaclient/utils.py", line 21, in <module>
    from oslo_serialization import jsonutils
  File "/Library/Python/2.7/site-packages/oslo_serialization/jsonutils.py", line 60, in <module>
    from oslo_utils import timeutils
  File "/Library/Python/2.7/site-packages/oslo_utils/timeutils.py", line 26, in <module>
    from monotonic import monotonic as now  # noqa
  File "build/bdist.macosx-10.11-intel/egg/monotonic.py", line 148, in <module>
    raise RuntimeError('no suitable implementation for this system')
RuntimeError: no suitable implementation for this system
`

Am I missing something monotonic needs, or is this a python-novaclient issue?

I'm Running on OS X 10.11 El Capitan with Python 2.7.10.

Regards,
Lennart

**Comment by lweijl** [comment]
This seems to be related to the following. https://georgegarside.com/blog/osx/package-incompatible-installer/

Disabling System Integrity Protection seems to fix the issue, not sure it's ideal though.

**Comment by vegaed** [comment]
Just ran into the same issue.

**Comment by miroswan** [comment]
I ran into this issue when running neutron. What I don't understand is why this is specific to the monotonic package. I didn't get errors on the other packages that were software dependencies.

**Comment by miroswan** [comment]
I think this is the problem regarding: https://github.com/atdt/monotonic/blob/master/monotonic.py#L72

```
GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ctypes
>>> ctypes.CDLL('libc.dylib', use_errno=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ctypes/__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(libc.dylib, 6): image not found
```

**Comment by miroswan** [comment]
Passing the full path works:

```
ython 2.7.10 (default, Aug 22 2015, 20:33:39)
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ctypes
>>> ctypes.CDLL('/usr/lib/libc.dylib', use_errno=True)
<CDLL '/usr/lib/libc.dylib', handle 7fff6154ca88 at 10ed5aa50>
```

**Comment by miroswan** [comment]
I have a fix coming soon.

**Comment by miroswan** [comment]
Just waiting on a merge and deployment: https://github.com/atdt/monotonic/pull/11

**Comment by atdt** [comment]
Fixed by @miroswan in #11. Thanks!
